workers/dockerfiles/Dockerfile.salmon

```diff
@@ -23,41 +30,29 @@ WORKDIR /home/user
 
 # Install Salmon
 ENV SALMON_VERSION 0.9.1
-
 RUN wget https://github.com/COMBINE-lab/salmon/releases/download/v${SALMON_VERSION}/Salmon-${SALMON_VERSION}_linux_x86_64.tar.gz
-
 RUN tar -xzf Salmon-${SALMON_VERSION}_linux_x86_64.tar.gz
-
-# Salmon can extract to a different directory than the name of the tar file.
-RUN cp `tar -tzf Salmon-${SALMON_VERSION}_linux_x86_64.tar.gz | head -1 | cut -f1 -d"/"`/bin/salmon /usr/local/bin
-
-RUN cp `tar -tzf Salmon-${SALMON_VERSION}_linux_x86_64.tar.gz | head -1 | cut -f1 -d"/"`/lib/* /usr/local/lib
-
-RUN rm -r Salmon*
+RUN rm -f Salmon-${SALMON_VERSION}_linux_x86_64.tar.gz
 # End Salmon installation.
 
-# Package list based on:
-# https://github.com/COMBINE-lab/salmon/blob/master/docker/Dockerfile
-ENV PACKAGES gcc make g++ cmake libboost-all-dev liblzma-dev libbz2-dev \
-    ca-certificates zlib1g-dev unzip autoconf
-
-RUN apt-get update && \
-    apt-get install -y --no-install-recommends ${PACKAGES} && \
-    apt-get clean
+# Install tximport
+COPY workers/install_tximport.R .
+RUN Rscript install_tximport.R
 
+# Install SalmonTools
 RUN git clone https://github.com/COMBINE-lab/SalmonTools.git
 RUN cd SalmonTools && cmake . -DCMAKE_INSTALL_PREFIX=/usr/local && make install
 
 RUN pip3 install --upgrade pip
 
 # We need JRE for QC tools
-RUN apt-get update -y
-RUN apt-get install -y unzip default-jre --fix-missing
+RUN apt-get update -y && apt-get install -y default-jre --fix-missing
```

Comment:
I removed `unzip` here because it is already installed implicitly as a dependency of `r-base`.